zerver/tests/test_auth_backends.py

```diff
@@ -2698,7 +2828,8 @@ def test_login_success_with_username(self) -> None:
     @override_settings(AUTHENTICATION_BACKENDS=('zproject.backends.ZulipLDAPAuthBackend',))
     def test_login_success_with_email_attr(self) -> None:
         with self.settings(LDAP_EMAIL_ATTR='mail'):
-            user_profile = self.backend.authenticate(username=self.ldap_username("aaron"),
+            user_profile = self.backend.authenticate(request=mock.MagicMock(),
+                                                     username=self.ldap_username("aaron"),
                                                      password=self.ldap_password(),
                                                      realm=get_realm('zulip'))
```

Comment:
This test class should possibly be rewritten to use `self.login` attempts rather than explicit `authenticate()` calls; I don't think the approach of having a bunch of tests passing in a MagicMock() is very clean.  Can be a follow-up.